[FLINK-21101][Azure] Add nightly profile to run all tests with the declarative scheduler

rmetzger opened the pull request:
## What is the purpose of the change

Get test exposure for decl scheduler

## Brief change log

- see commit history

The change to `azure-pipelines.yml` is just for review / demo purposes.

zentol commented on flink-test-utils-parent/flink-test-utils-junit/src/main/java/org/apache/flink/testannotations/FailsWithDeclarativeScheduler.java:
```diff
@@ -0,0 +1,22 @@
+ * limitations under the License.
+ */
+
+package org.apache.flink.testannotations;
```
org.apache.flink.testutils.junit

zentol commented on flink-end-to-end-tests/run-nightly-tests.sh:
```diff
@@ -57,6 +57,11 @@ if [ ! -z "$TF_BUILD" ] ; then
 	on_exit run_on_exit
 fi
 
+if [[ ${PROFILE} == *"enable-declarative-scheduler"* ]]; then
```
we also need to disable the kubernetes/yarn e2e tests; forgot to create those tickets.

zentol commented on flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/JobMasterQueryableStateTest.java:
```diff
@@ -67,6 +69,7 @@
 import static org.junit.Assert.fail;
 
 /** Tests for the queryable-state logic of the {@link JobMaster}. */
+@Category(FailsWithDeclarativeScheduler.class)
```
This one should no longer fail.

zentol commented on flink-tests/src/test/java/org/apache/flink/test/streaming/runtime/TimestampITCase.java:
```diff
@@ -159,6 +161,7 @@ public void testWatermarkPropagation() throws Exception {
     }
 
     @Test
+    @Category(FailsWithDeclarativeScheduler.class)
```
missing reference to FLINK-21333

zentol commented on tools/azure-pipelines/build-apache-repo.yml:
```diff
@@ -157,6 +157,17 @@ stages:
+          environment: PROFILE="-Dhadoop.version=2.8.3 -Dinclude_hadoop_aws -Dscala-2.11 -Penable-declarative-scheduler"
+          run_end_to_end: true
+          container: flink-build-container
+          jdk: jdk11
```
```suggestion
          jdk: jdk8
```

rmetzger commented:
Thanks a lot for your review. I addressed your comments, and ignored a few more tests, since there are still failures: https://dev.azure.com/rmetzger/Flink/_build/results?buildId=8902&view=results
I'll ping you for a final look once the profile is green.

zentol commented on flink-end-to-end-tests/run-nightly-tests.sh:
```diff
@@ -194,10 +194,12 @@ run_test "Batch SQL end-to-end test" "$END_TO_END_DIR/test-scripts/test_batch_sq
-run_test "Streaming File Sink s3 end-to-end test" "$END_TO_END_DIR/test-scripts/test_file_sink.sh s3 StreamingFileSink" "skip_check_exceptions"
-run_test "New File Sink end-to-end test" "$END_TO_END_DIR/test-scripts/test_file_sink.sh local FileSink" "skip_check_exceptions"
-run_test "New File Sink s3 end-to-end test" "$END_TO_END_DIR/test-scripts/test_file_sink.sh s3 FileSink" "skip_check_exceptions"
+if [[ ${PROFILE} != *"enable-declarative-scheduler"* ]]; then
```
This is probably due to FLINK-21400

rmetzger commented:
The tests have finally passed: https://dev.azure.com/rmetzger/Flink/_build/results?buildId=8908&view=results
I now rebased to the latest master, did the renaming again. If CI has passed tomorrow, I'll merge the change

zentol commented on azure-pipelines.yml:
```diff
@@ -71,7 +71,7 @@ stages:
           e2e_pool_definition:
             vmImage: 'ubuntu-16.04'
-          environment: PROFILE="-Dhadoop.version=2.8.3 -Dinclude_hadoop_aws -Dscala-2.11"
+          environment: PROFILE="-Dhadoop.version=2.8.3 -Dinclude_hadoop_aws -Dscala-2.11 -Penable-adaptive-scheduler"
```
remember to revert this ;)

zentol commented on flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/JobMasterTest.java:
```diff
@@ -1044,6 +1050,7 @@ public void testResourceManagerConnectionAfterStart() throws Exception {
      * if this execution fails.
      */
     @Test
+    @Category(FailsWithAdaptiveScheduler.class) // FLINK-21399
```
instead we could reference FLINK-21450; then I don't have to touch this test when I address FLINK-21399 🙈

rmetzger commented:
Will close the Jira ticket, once Jira is available again.
